python/cuml/cuml/fil/fil.pyx

```diff
@@ -771,9 +775,10 @@ class ForestInference(Base, CMajorInputTagMixin):
             pool to use during loading and inference.
         """
         if threshold is not None:
-            raise ValueError(
-                "`load()` no longer accepts `threshold` parameter. "
-                "Set `threshold` in `predict()` instead."
+            warnings.warn(
+                "The `threshold` parameter of `load()` is deprecated and "
+                "will be removed in 25.10. Please set `threshold` in `predict()` instead.",
+                FutureWarning
```

Comment:
Let's make sure to update the docs to clarify that this is not technically a `DeprecationWarning`. I would suggest to just refer to is as "deprecation warning" (without backticks).

https://github.com/rapidsai/cuml/blob/1bd2c5bca4f38f8eadbfc87d9ad03d9236a2de73/docs/source/FIL.rst?plain=1#L36

It should say: "will trigger a deprecation warning" rather than "raise a `DeprecationWargning`"